lib/sycamore/sycamore/transforms/aryn_partitioner.py

```diff
@@ -0,0 +1,50 @@
+from typing import BinaryIO
+from collections.abc import Mapping
+import requests
+import json
+from sycamore.data.element import create_element
+
+_DEFAULT_ARYN_PARTITIONER_ADDRESS = "https://api.aryn.cloud:8000/v1/partition"
+
+
+class ArynPDFPartitionerException(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+class ArynPDFPartitioner:
+    @staticmethod
+    def partition_pdf(
+        file: BinaryIO,
+        token: str,
+        aryn_partitioner_address=_DEFAULT_ARYN_PARTITIONER_ADDRESS,
+        threshold: float = 0.4,
+        use_ocr: bool = False,
+        ocr_images: bool = False,
+        ocr_tables: bool = False,
+        extract_table_structure: bool = False,
+        extract_images: bool = False,
+    ):
+        options = {
+            "threshold": threshold,
+            "use_ocr": use_ocr,
+            "ocr_images": ocr_images,
+            "ocr_tables": ocr_tables,
+            "extract_table_structure": extract_table_structure,
+            "extract_images": extract_images,
+        }
+        files: Mapping = {"pdf": file, "options": json.dumps(options).encode("utf-8")}
+        header = {"Authorization": f"Bearer {token}"}
+        response = requests.post(aryn_partitioner_address, files=files, headers=header)
+
+        if response.status_code != 200:
```

Comment:
should we retry depending on the error code? Also, what does a failure mean for sycamore pipeline?

Reply:
Yes. We should retry when we get an error saying try again later.  I'll add this.